From 92edc5c8089196c36bc19d09376dc38eca71b3fe Mon Sep 17 00:00:00 2001
From: Dmitry Kazakov <dimula73@gmail.com>
Date: Thu, 16 Feb 2023 14:24:08 +0300
Subject: [PATCH] Fix compilation with clang-15

std::unary_function and std::binary_function are removed from the
standard.
---
 boost/functional.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/boost/functional.hpp b/boost/functional.hpp
index 644307847..27d36710e 100644
--- a/boost/functional.hpp
+++ b/boost/functional.hpp
@@ -21,7 +21,7 @@ namespace boost
     namespace functional
     {
         namespace detail {
-#if defined(_HAS_AUTO_PTR_ETC) && !_HAS_AUTO_PTR_ETC
+#if (defined(_HAS_AUTO_PTR_ETC) && !_HAS_AUTO_PTR_ETC) || defined(BOOST_NO_CXX98_FUNCTION_BASE)
             // std::unary_function and std::binary_function were both removed
             // in C++17.
 
-- 
2.38.1.windows.1

